src/components/renderer/renderer.tsx

```diff
@@ -126,15 +127,27 @@ class Editor extends React.PureComponent<Props, State> {
   }
 
   public initView() {
-    const {vegaSpec, vegaLiteSpec, normalizedVegaLiteSpec, config, baseURL, mode, setView, setRuntime, hoverEnable} =
-      this.props;
+    const {
+      vegaSpec,
+      vegaLiteSpec,
+      normalizedVegaLiteSpec,
+      config,
+      baseURL,
+      mode,
+      setView,
+      setRuntime,
+      hoverEnable,
+      expressionInterpreter,
+    } = this.props;
+
+    const parseOptions = expressionInterpreter ? {ast: true} : {};
 
     let runtime: vega.Runtime;
     if (mode === Mode.VegaLite) {
       // In vl mode, we compile Vega-Lite spec along with config to Vega spec
-      runtime = vega.parse(vegaSpec);
+      runtime = vega.parse(vegaSpec, {}, parseOptions);
     } else {
-      runtime = vega.parse(vegaSpec, config as VgConfig);
+      runtime = vega.parse(vegaSpec, config as VgConfig, parseOptions);
     }
```

Comment:
Could probably simplify this to `mode === Mode.VegaLite ? {} : config`

Reply:
Good idea, just pushed the change 👍